sshfs.c

```diff
@@ -190,6 +204,9 @@ struct sshfs_file {
 	int connver;
 	int modifver;
 	int refs;
+#ifdef __APPLE__
+	pthread_mutex_t file_lock;
```

Comment:
I don't know and this may be hard to track down.

I see this comes from b4023a19dd7ec7a099d2e0df491547cf3bb6bec3, which is a very old commit that says nothing more than `Apply modified sshfs 2.2 patch from MacFUSE`. I guess the locking semantics in MacFUSE might have been different at the time (though that'd have been stupid for compatibility).

I've removed this for now assuming it's a hack that's no longer necessary. I have ran a few local tests and things seem stable. Those tests are not very scientific... but sshfs has no test suite anyway. I can try a much stronger workload tomorrow at work to better assess this.